saleor/graphql/discount/mutations/voucher/voucher_code_bulk_delete.py

```diff
@@ -0,0 +1,96 @@
+import graphene
+
+from .....core.tracing import traced_atomic_transaction
+from .....discount import models
+from .....permission.enums import DiscountPermissions
+from .....webhook.event_types import WebhookEventAsyncType
+from ....core.descriptions import ADDED_IN_318
+from ....core.enums import VoucherCodeBulkDeleteErrorCode
+from ....core.mutations import BaseMutation
+from ....core.types import NonNullList, VoucherCodeBulkDeleteError
+from ....core.utils import WebhookEventInfo
+from ....plugins.dataloaders import get_plugin_manager_promise
+from ...types import VoucherCode
+
+
+class VoucherCodeBulkDelete(BaseMutation):
+    count = graphene.Int(
+        required=True, description="Returns how many codes were deleted."
+    )
+
+    class Arguments:
+        ids = NonNullList(
+            graphene.ID,
+            required=True,
+            description="List of voucher codes IDs to delete.",
+        )
+
+    class Meta:
+        description = "Deletes voucher codes." + ADDED_IN_318
+        model = models.VoucherCode
+        object_type = VoucherCode
+        permissions = (DiscountPermissions.MANAGE_DISCOUNTS,)
+        error_type_class = VoucherCodeBulkDeleteError
+        webhook_events_info = [
+            WebhookEventInfo(
+                type=WebhookEventAsyncType.VOUCHER_UPDATED,
+                description="A voucher was updated.",
+            )
+        ]
+
+    @classmethod
+    def clean_codes(cls, codes, errors_list):
+        invalid_codes_ids = []
+        cleaned_ids = set()
+
+        for index, code in enumerate(codes):
+            obj_type, code_pk = graphene.Node.from_global_id(code)
+            if obj_type != "VoucherCode":
+                invalid_codes_ids.append(code)
+                continue
+
+            cleaned_ids.add(code_pk)
+
+        if invalid_codes_ids:
+            errors_list.append(
+                VoucherCodeBulkDeleteError(
+                    path=f"ids.{index}",
```

Comment:
Index here won't be the last value from iteration above? It's expected?

Reply:
This code could result in `NameError` exception in case where variable `codes` would be empty as variable `index` would not be created.